### PR TITLE
Backport typography block supports for WP 6.1 (refresh)

### PR DESCRIPTION
**Pull request by ironprogrammer**
Refresh of #3203.

- Fixes 2 failing unit tests (depends on blocking #3199).
- Updates DocBlock formatting to [WP standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods).

From [original PR](https://github.com/WordPress/wordpress-develop/pull/3203):

> ❗ BLOCKED by https://github.com/WordPress/wordpress-develop/pull/3199
> 
> Backporting typography block supports changes from Gutenberg to WP 6.1
> 
> See tracking issue: https://github.com/WordPress/gutenberg/issues/43440
> 
> ### Gutenberg sync PR
> 
> Once this PR lands we can merge https://github.com/WordPress/gutenberg/pull/43928
> 
> ### Main changes
> 
> - https://github.com/WordPress/gutenberg/pull/39529
> - https://github.com/WordPress/gutenberg/pull/40332
> 
> Trac ticket: https://core.trac.wordpress.org/ticket/56467

Props @ramonjd.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by ockham** [comment]
@ironprogrammer Thank you! The [style engine PR](https://github.com/WordPress/wordpress-develop/pull/3199) has now been merged; would you mind rebasing this PR? 😊

**Comment by ironprogrammer** [comment]
@ockham Done!

CC @ramonjd.

**Comment by ironprogrammer** [comment]
Hmmm, the test failures may be related to #3199, which removes spaces between generated CSS attributes and values. Looking into it.

However, the tests in this PR are passing.

**Comment by andrewserong** [comment]
> Hmmm, the test failures may be related to https://github.com/WordPress/wordpress-develop/pull/3199, which removes spaces between generated CSS attributes and values.

@ironprogrammer I think in this case I'd lean toward updating those existing tests. As you mention. the style engine intentionally defaults to not prettifying output, so inline styles therefore don't have spaces. Given that there'll be a slight mismatch until all of the block supports are merged, I wonder if the `assert_content_and_styles_and_classes_match` test function should be updated to be more permissive of spaces? (It currently normalizes spaces between property/value pairs, but not between the `colon` and the value here: https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/blocks/supportedStyles.php#L155)

**Comment by ramonjd** [comment]
Thank for you for advancing this one!! 🙇

**Comment by hellofromtonya** [comment]
Currently testing and reviewing. When ready, will do the commit.

File: src/wp-includes/block-supports/typography.php
Comment (hellofromtonya) [inline src/wp-includes/block-supports/typography.php]:
```suggestion
 * @since 6.1.0 Used the style engine to generate CSS and classnames.
```

File: tests/phpunit/tests/block-supports/typography.php
Comment (hellofromtonya) [inline tests/phpunit/tests/block-supports/typography.php]:
```suggestion
	 * @param string $expected_output             Expected output.
```

File: tests/phpunit/tests/block-supports/typography.php
Comment (hellofromtonya) [inline tests/phpunit/tests/block-supports/typography.php]:
```suggestion
	/**
	 * Data provider.
	 *
	 * @return array
	 */
```

**Review by hellofromtonya — APPROVED** [review]
- Meets coding standards ✅ 
- Merge conflict fixed ✅ 
- Test passing ✅ 

Ready for commit.

**Comment by hellofromtonya** [comment]
Prepping commit now.

File: src/wp-includes/block-supports/typography.php
Comment (hellofromtonya) [inline src/wp-includes/block-supports/typography.php:361]: Why is this needed twice? @ockham

Comment (ockham) [reply src/wp-includes/block-supports/typography.php:361]: TBH I know next to nothing about this code -- I've only rebased it 😅 

Speaking of rebase, maybe the duplicated code is a result of a slightly wonky rebase?

Comment (hellofromtonya) [reply src/wp-includes/block-supports/typography.php:361]: Nope, it's the original Gutenberg PR https://github.com/WordPress/gutenberg/pull/39529/.

Hey @ramonjd is this duplicate code needed?

Comment (ockham) [reply src/wp-includes/block-supports/typography.php:361]: > Nope, it's the original Gutenberg PR https://github.com/WordPress/gutenberg/pull/39529.

Ah, I hadn't noticed! Huh, weird.

(If it's blocking this from merging, I'd go out on a limb and say that we might remove it? 😅 )

Comment (ockham) [reply src/wp-includes/block-supports/typography.php:361]: Hmm, it's been duplicated for a while: https://github.com/WordPress/gutenberg/commit/eea3aebb1ab4de6fa300bd24bfb1f7afa3b8051e#diff-2c54d3c8305590cf826f83511a9fd85d5b405470addd62edc3183ff9118177fbL302

Comment (hellofromtonya) [reply src/wp-includes/block-supports/typography.php:361]: I'm going to remove it. If it's an issue, a Trac ticket can be opened to fix it during the beta cycle.

Comment (hellofromtonya) [reply src/wp-includes/block-supports/typography.php:361]: Scratch that - my bad 🤦‍♀️ The 2nd time is for the minimum. Sorry for the confusion. Leaving it in (and smacking my forehead 🤣 ).

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/54260. Thank you everyone for contributions!

File: src/wp-includes/block-supports/typography.php
Comment (ramonjd) [reply src/wp-includes/block-supports/typography.php:361]: Thanks for the ping, folks!

Yep, one is for `$minimum_viewport_width` and the other for `$maximum_viewport_width`. 

If you think the vars could be named more clearly feel free to change. I can sync in Gutenberg today.

**Pull request closed without merging**
